src/components/cart/CheckRecipe.tsx

```diff
@@ -0,0 +1,136 @@
+import { Box, Flex, Image, Text } from "@chakra-ui/react";
+import { AddIcon, MinusIcon, SmallCloseIcon } from "@chakra-ui/icons";
+import { FC } from "react";
+import { useMyStore } from "../store/store.tsx";
+import { NewRecipe, Product } from "../types.ts";
+import { useGetRecipePrice } from "../hooks/useGetRecipePrice.tsx";
+
+type Props = {
+  recipeInCart: string;
+};
+
+type IngredientData = null | {
+  productsByStoreId: {
+    [storeId: string]: Product[];
+  };
+  ingredientIds: {
+    [storeId: string]: string[];
+  };
+};
+
+const CheckRecipe: FC<Props> = ({ recipeInCart }) => {
+  const { recipes } = useMyStore();
+  const findRecipeById = (recipeInCart: string) =>
```

Comment:
I vzhledem k tomu, ze tohle pouzivas na vice mistech (ale duplikujes to), tak navrhuju dat do do toho RecepiesSlice a pak to pouzit z useMyStore. 

Vyhoda je, ze i ta reference (na funkci) zustane stejna. Tady to v kazem renderu bude "jina" fuknce. Popravde nevim jak moc se citis silna v kramflecich, co se referenci v javascriptu tyce. Dan Abramov to ma skvele zpracovane v jeho "knizce" https://justjavascript.com/. Kazdopadne to, ze je to kazdy render jina funkce (nejjednodussi reseni by bylo vytahnout jit mimo tu komponentu (ale pak bys ji musela predavat jako argument i `recipes`), tady v tomhle pripade moc nevadi.